torch/csrc/utils/tensor_new.cpp

```diff
@@ -476,20 +438,25 @@ Tensor legacy_sparse_tensor_new(c10::DispatchKey dispatch_key, at::ScalarType sc
       throw TypeError("SparseTensor.new(sequence) only accepts sizes.  Please use torch.sparse_coo_tensor() " \
                       "or construct a strided tensor and convert it to sparse via to_sparse.");
     }
-    return new_with_sizes(dispatch_key, scalar_type, r.deviceOptional(1), r.intlist(0));
+    return new_with_sizes(options, scalar_type, r.deviceOptional(1), r.intlist(0));
   }
   throw std::runtime_error("new(): invalid arguments");
 }
 
 // NB: device_idx here is NOT a DeviceIndex, but index into PythonArgs
-c10::DispatchKey typeIdWithDefault(PythonArgs& r, int64_t device_idx, c10::DispatchKey dispatch_key) {
-  auto device_type = r.isNone(device_idx) ? computeDeviceType(dispatch_key) : r.device(device_idx).type();
-  return backendToDispatchKey(backendToBackendOfDeviceType(dispatchKeyToBackend(dispatch_key), device_type));
+c10::TensorOptions typeIdWithDefault(PythonArgs& r, int64_t device_idx, c10::DispatchKey dispatch_key) {
+  auto options = dispatchKeyToTensorOptions(dispatch_key);
+  if (!r.isNone(device_idx)) {
+    // TODO: This line doesn't seem to be exercised at all in tests
+    options = options.device(r.device(device_idx).type());
```

Comment:
Doesn't this still have the problem you described in the "janky" note? Index gets scraped off in the same way as before I think.

Reply:
Yes, I think you're right. It's too bad I can't tell if it matters or not cuz it's not exercised by tests LOL